sched/group/group_killchildren.c

```diff
@@ -110,7 +110,7 @@ static int group_cancel_children_handler(pid_t pid, FAR void *arg)
 
           if ((rtcb->flags & TCB_FLAG_TTYPE_MASK) == TCB_FLAG_TTYPE_PTHREAD)
             {
-              ret = pthread_cancel(pid);
```

Comment:
I'm fairly certain this will not fly. There are three types of task exit:

- Normal termination, where I think pthread_cancel is fine
- Abnormal termination, where I also think pthread_cancel is fine
- Fault termination, where pthread_cancel should not be used

This modification now changes the behavior for all 3 use cases.

I think we do need a separate "group_terminate_children_handler" or something that will do 99% same work, but instead of trying to cancel the thread nicely, just murders it.

Reply:
right, removed this